src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1141,8 +1137,13 @@ def dbapi(
 
     def _infer_data_source_schema(
         self, conn: Connection, table: str
-    ) -> Tuple[StructType, Tuple[Tuple[str, Any, int, int, int, int, bool]]]:
-        raw_schema = conn.execute(f"SELECT * FROM {table} WHERE 1 = 0").description
+    ) -> Tuple[StructType, Tuple[Tuple[str, Any, int, int, int, bool]]]:
+        query = f"""
+            SELECT COLUMN_NAME, DATA_TYPE, CHARACTER_MAXIMUM_LENGTH, NUMERIC_PRECISION, NUMERIC_SCALE, IS_NULLABLE
+            FROM INFORMATION_SCHEMA.COLUMNS
+            WHERE TABLE_NAME = '{table}'
+            """
+        raw_schema = conn.execute(query).fetchall()
```

Comment:
this only works for sql server right? so we'd better find ways to tell wether the target database is sql server,  otherwise we raise unsupported error.
you can check if pyodbc offers any api to get the info